controllers/chat/moveToAnon.js

```diff
@@ -58,8 +58,6 @@ const moveToAnon = async (req, res) => {
       (user) => user.user_id !== mySignedId
     )[0];
     targetUserId = prevTargetUser.user_id;
-  } else {
-    return res.status(404).json(responseError('Old channel id not found'));
   }
 
   if (!prevTargetUser) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of the 404 error response when `prevTargetUser` is not found could lead to potential issues. If `prevTargetUser` is null or undefined, the code will throw an error on line 60 when trying to access `user_id`. It's better to handle this case explicitly.
```diff
-    if (!prevTargetUser) {
+    if (!prevTargetUser) {
+      return res.status(404).json(responseError('Old channel id not found'));
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->